pkg/sources/adapter/salesforcesource/client/bayeux.go

```diff
@@ -162,6 +166,8 @@ func (b *bayeux) Start(ctx context.Context) error {
 	b.ctx = ctx
 	b.mutex.Unlock()
 
+	bom := wait.NewExponentialBackoffManager(time.Second, time.Minute*5, time.Minute*30, 2, 0, &clock.RealClock{})
```

Comment:
If I understand this right, there are can be scenarios when `ctx.Done` is fired but we're blocked for 5 minutes in the `default` branch with this backoff timer, leaving the pod unresponsive for say rolling updates? Or is it considered to be the edge case and we're okay with it?

Reply:
good point.

it is edgy, but also an anti-pattern. I cannot find  an ExponentialBackoffManager that accepts a context, but let me try to find a better solution.

Reply:
https://github.com/triggermesh/triggermesh/pull/332/commits/f4f6c40a387da83bd497674cec6adf12caed83fd

Can you take a look?

Reply:
That's rather elegant channels management 🙂 
If you tested it and there are no blocking conditions occurred, then lgtm!